framework/src/base/Assembly.C

```diff
@@ -4549,7 +4549,8 @@ Assembly::modifyFaceWeightsDueToXFEM(const Elem * elem, unsigned int side)
 void
 Assembly::hasScalingVector()
 {
-  _scaling_vector = &_sys.getVector("scaling_factors");
+  if (_sys.hasVector("scaling_factors"))
+    _scaling_vector = &_sys.getVector("scaling_factors");
```

Comment:
Hmm I would prefer to rather not call this method if the system doesn't have a scaling factor vector. It seems like if the system doesn't have a scaling factor vector, then calling this method should result in an error (which it would pre-this-PR)

Reply:
See what you think of [bb830af](https://github.com/idaholab/moose/pull/26282/commits/bb830af5e1897311078a7a6ff9955f42bc73c999)

Reply:
This is much better! Thanks a lot!